### DISCUSSION TITLE
Returning an error via Web API to unauthorised users?

### DISCUSSION THREAD

**Discussion by uxsamok** [pocketbase/pocketbase#3990] [Q&A] [answered]
Is there a way to configure PocketBase so that it returns an error code instead of a 200 response code when the authorization bearer token is invalid?

Currently, when I use the API Rule “_id = @request.auth.id_” and send a List request with an invalid token, Web API responds with code 200 and an empty array: 

`{"page":1,"perPage":30,"totalItems":0,"totalPages":0,"items":[]}`.

That results challenge with WeWeb, where the bearer authentication process depends on receiving an error code for invalid tokens. If WeWeb gets a 200 code with an empty result, it interprets it as a successful user fetch, mistakenly granting access to private pages.

Any guidance on how to ensure PocketBase returns an error code for invalid tokens via Web API would be greatly appreciated.
I'm probably missing an obvious solution.

**Comment by jonshipman** [comment, marked as answer]
Try this https://pocketbase.io/docs/js-event-hooks/#onrecordslistrequest

When the user is not authenticated, you can `throw new ApiError(403, 'Forbidden');`

```
const requestInfo = $apis.requestInfo(e.httpContext);
if (!requestInfo.authRecord) throw new ApiError(403, 'Forbidden');
```

**Reply by uxsamok** [reply]
Thanks so much for the help! That was exactly what I needed. 

If anybody needs to use PockeBase with WeWeb, here is my full code based on jonshipman's reply.
You need to add it to a file with anyName.pb.js and put in the pb/pb_hooks/ folder. 

The code creates a custom route _/api/custom/users/records_ that can be used as a _"User Endpoint"_ for _Token Based Auth_ plugin in WeWeb. The route is accepting the authorisation token and

- if the token is valid, returns an array with one item describing the user. 
- if the token is not valid, returns 403 error (WeWeb needs to receive an error to be able to log out the user)

```
routerAdd("GET", "/api/custom/users/records", (c) => {
    const requestInfo = $apis.requestInfo(c);

    if (!requestInfo.authRecord) {
        throw new ApiError(403, 'Forbidden');
    }

    try {
        const records = $app.dao().findRecordsByFilter(
            "users",
            "id = '" + requestInfo.authRecord.id + "'"
        );

        return c.json(200, records);
    } catch (error) {
        throw new ApiError(500, 'Error fetching records');
    }
});
```
